gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -789,119 +795,261 @@ local function FindCorpse(ply)
 	end
 end
 
-local poss = {}
+-- Handles all stuff needed if the revival failed
+local function OnReviveFailed(ply, failMessage)
+	if isfunction(ply.OnReviveFailedCallback) then
+		ply.OnReviveFailedCallback(ply, failMessage)
 
--- Populate Around Player
-for i = 0, 360, 22.5 do
-	poss[#poss + 1] = Vector(math.cos(i), math.sin(i), 0)
-end
-
-poss[#poss + 1] = Vector(0, 0, 1) -- Populate Above Player
-
-local function FindCorpsePosition(corpse)
-	local size = Vector(32, 32, 72)
-	local startPos = corpse:GetPos() + Vector(0, 0, size.z * 0.5)
-	local len = #poss
-
-	for i = 1, len do
-		local v = poss[i]
-		local pos = startPos + v * size * 1.5
-
-		local tr = {}
-		tr.start = pos
-		tr.endpos = pos
-		tr.mins = size * -0.5
-		tr.maxs = size * 0.5
-
-		local trace = util.TraceHull(tr)
-
-		if not trace.Hit then
-			return pos - Vector(0, 0, size.z * 0.5)
-		end
+		ply.OnReviveFailedCallback = nil
+	else
+		LANG.Msg(ply, failMessage, nil, MSG_MSTACK_WARN)
 	end
 
-	return false
+	net.Start("TTT2RevivalStopped")
+	net.Send(ply)
 end
 
 ---
 -- Revives a @{Player}
--- @param number delay the delay of the revive
--- @param function fn the @{function} that should be run if the @{Player} revives
--- @param function check an additional checking @{function}
--- @param boolean needcorpse whether the dead @{Player} CORPSE is needed
--- @param boolean force if the @{Player} revive is already forced. Useful if you have multiple reviving equipments
--- @param function onFail this @{function} is called if the revive fails
--- @realm server
-function plymeta:Revive(delay, fn, check, needcorpse, force, onFail)
+-- @param [default=3]number delay The delay of the revive
+-- @param [opt]function OnRevive The @{function} that should be run if the @{Player} revives
+-- @param [opt]function DoCheck An additional checking @{function}
+-- @param [default=false]boolean needsCorpse Whether the dead @{Player} @{CORPSE} is needed
+-- @param [default=false]boolean blockRound Stops the round from ending if this is set to true until the player is alive again
+-- @param [opt]function OnFail This @{function} is called if the revive fails
+-- @param [opt]Vector spawnPos The position where the player should be spawned, accounts for minor obstacles
+-- @param [opt]Angle spawnEyeAngle The eye angles of the revived players
+-- @realm server
+function plymeta:Revive(delay, OnRevive, DoCheck, needsCorpse, blockRound, OnFail, spawnPos, spawnEyeAngle)
 	local ply = self
```

Comment:
This isn't needed here

Reply:
This is from you^^

Reply:
Yea we came up localizing `self` before using it in timer, but this isn't needed outside the timer. But you are right